receiver/mongodbatlasreceiver/internal/metadata/metric_name_mapping.go

```diff
@@ -15,707 +15,676 @@
-		"object_type": pdata.NewValueString("collection"),
-	}},
-	"DATABASE_INDEX_COUNT": {"mongodbatlas.db.counts", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("index"),
-	}},
-	"DATABASE_EXTENT_COUNT": {"mongodbatlas.db.counts", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("extent"),
-	}},
-	"DATABASE_OBJECT_COUNT": {"mongodbatlas.db.counts", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("object"),
-	}},
-	"DATABASE_VIEW_COUNT": {"mongodbatlas.db.counts", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("view"),
-	}},
-	"DATABASE_AVERAGE_OBJECT_SIZE": {"mongodbatlas.db.size", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("object"),
-	}},
-	"DATABASE_STORAGE_SIZE": {"mongodbatlas.db.size", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("storage"),
-	}},
-	"DATABASE_INDEX_SIZE": {"mongodbatlas.db.size", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("index"),
-	}},
-	"DATABASE_DATA_SIZE": {"mongodbatlas.db.size", map[string]pdata.Value{
-		"object_type": pdata.NewValueString("data"),
-	}},
+	"DATABASE_COLLECTION_COUNT": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Collection)
+	},
+	"DATABASE_INDEX_COUNT": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Index)
+	},
+	"DATABASE_EXTENT_COUNT": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Extent)
+	},
+	"DATABASE_OBJECT_COUNT": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.Object)
+	},
+	"DATABASE_VIEW_COUNT": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbCountsDataPoint(ts, float64(*dp.Value), AttributeObjectType.View)
+	},
+	"DATABASE_AVERAGE_OBJECT_SIZE": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Object)
+	},
+	"DATABASE_STORAGE_SIZE": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Storage)
+	},
+	"DATABASE_INDEX_SIZE": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Index)
+	},
+	"DATABASE_DATA_SIZE": func(mb *MetricsBuilder, dp *mongodbatlas.DataPoints, ts pdata.Timestamp) {
+		mb.RecordMongodbatlasDbSizeDataPoint(ts, float64(*dp.Value), AttributeObjectType.Data)
+	},
```

Comment:
I'm good with this approach, but a switch case construction would probably be cleaner than using a map. WDYT?

Reply:
Yeah I agree. I was trying to keep the code as close to the original as possible but I don't think a switch case is a huge change and it is cleaner. I'll make the change.

Reply:
@dmitryax updated!